streams-contrib/streams-persist-graph/src/main/java/org/apache/streams/graph/GraphHttpPersistWriter.java

```diff
@@ -84,54 +86,79 @@ else if( configuration.getType().equals(GraphHttpConfiguration.Type.REXSTER)) {
+            activityObject = (ActivityObject) entry.getDocument();
+        } else {
+            ObjectNode objectNode;
+            if (entry.getDocument() instanceof ObjectNode) {
+                objectNode = (ObjectNode) entry.getDocument();
+            } else if( entry.getDocument() instanceof String) {
+                try {
+                    objectNode = mapper.readValue((String) entry.getDocument(), ObjectNode.class);
+                } catch (IOException e) {
+                    LOGGER.error("Can't handle input: ", entry);
+                    return mapper.createObjectNode();
+                }
+            } else {
+                LOGGER.error("Can't handle input: ", entry);
+                return mapper.createObjectNode();
+            }
+
+            if( objectNode.get("verb") != null ) {
+                try {
+                    activity = mapper.convertValue(objectNode, Activity.class);
+                } catch (Exception e) {
+                    activityObject = mapper.convertValue(objectNode, ActivityObject.class);
+                }
+            } else {
+                activityObject = mapper.convertValue(objectNode, ActivityObject.class);
             }
         }
 
-        Preconditions.checkNotNull(activity);
+        Preconditions.checkArgument(activity != null || activityObject != null);
 
         ObjectNode request = mapper.createObjectNode();
         ArrayNode statements = mapper.createArrayNode();
 
-        activity.getActor().setObjectType("page");
-
         // always add vertices first
 
-        List<String> labels = Lists.newArrayList();
-        if( activity.getProvider() != null &&
-                !Strings.isNullOrEmpty(activity.getProvider().getId()) ) {
-            labels.add(activity.getProvider().getId());
-        }
+        List<String> labels = Lists.newArrayList("streams");
 
-        if( activity.getActor() != null &&
-                !Strings.isNullOrEmpty(activity.getActor().getId()) ) {
-            if( activity.getActor().getObjectType() != null )
-                labels.add(activity.getActor().getObjectType());
-            statements.add(httpGraphHelper.createHttpRequest(queryGraphHelper.mergeVertexRequest(activity.getActor())));
+        if( activityObject != null ) {
+            if (activityObject.getObjectType() != null)
+                labels.add(activityObject.getObjectType());
+            statements.add(httpGraphHelper.createHttpRequest(queryGraphHelper.mergeVertexRequest(activityObject)));
         }
 
-        if( activity.getObject() != null &&
-                !Strings.isNullOrEmpty(activity.getObject().getId()) ) {
-            if( activity.getObject().getObjectType() != null )
-                labels.add(activity.getObject().getObjectType());
-            statements.add(httpGraphHelper.createHttpRequest(queryGraphHelper.mergeVertexRequest(activity.getObject())));
-        }
+        if( activity != null ) {
+            if( activity.getProvider() != null &&
```

Comment:
More style points, similar comment as before, basically -- I'd save `provider`, `actor`, `object` variables here, I think would be more readable than calling `activity.getProvider()` etc in what follows.